src/main/java/ca/concordia/soen/ThrowsGenericVisitor.java

```diff
@@ -0,0 +1,72 @@
+package ca.concordia.soen;
+
+import org.eclipse.jdt.core.dom.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ThrowsGenericVisitor  extends ASTVisitor {
+    private final CompilationUnit compilationUnit;
+    int AntiPatternOccurrencesCount = 0;
+    List<AntiPatternOccurrence> ThrowsGenericOcurrencesList = new ArrayList<>();
+
+    public ThrowsGenericVisitor(CompilationUnit compilationUnit) {
+        this.compilationUnit = compilationUnit;
+    }
+
+    @Override
+    public boolean visit(MethodDeclaration node) {
+        List<Type> thrownExceptions = node.thrownExceptionTypes();
```

Comment:
This only capture the throws in the method declaration, right? It took me a while to understand this (and when the condition in line 27 would be satisfied). Maybe we could rename this var to methodDeclarationThrownExceptions to make it clearer to the reader. What do you think? :)

Reply:
Changed the variable name